deepchem/utils/openvino_model.py

```diff
@@ -71,27 +98,53 @@ def __init__(self,
     self._batch_size = batch_size
     self._outputs: List = []
 
+    self.input_size: List = []
+    self.last_input_shapes: List = []
+    self.net: IENetwork = None
+
   def _read_torch_model(self, generator: Iterable[Tuple[Any, Any, Any]]):
     """
     Prepare PyTorch model for OpenVINO inference:
       1. Convert into ONNX format
       2. Read network from memory
     NOTE: We do not load model in __init__ method because of training.
     """
-    import torch
+    # import dgl
+    import torch.nn.functional as F
 
     # We need to serialize ONNX model with real input shape.
     # So we create a copy of the generator to take first input.
     generator_copy, generator = itertools.tee(generator, 2)
+
+    # list with array of objects deepchem.feat.graph_data.GraphData (len 4)
     inputs, _, _ = next(generator_copy)
-    assert (len(inputs) == 1), 'Not implemented'
-    inp_shape = list(inputs[0].shape)
-    inp_shape[0] = self._batch_size
+
+    if (type(self._torch_model).__name__ == 'CGCNNModel'):
+      # Create Graph object
+      inputs, _, _ = self._torch_model._prepare_batch((inputs, None, None))
+      self._torch_model.model.graph = inputs
+      self._torch_model.model = CGCNN_OV(self._torch_model.model)
+
+      node_feats = inputs.ndata.pop('x')
+      edge_feats = inputs.edata.pop('edge_attr')
+      inp = [node_feats, edge_feats]
+    else:
+      assert (len(inputs) == 1), 'Not implemented'
+
+      inp_shape = list(inputs[0].shape)
+      inp_shape[0] = self._batch_size
+      inp = torch.randn(inp_shape)
+
+    # There is a bug in OpenVINO for Softplus
+    # It will be fixed with this changes: https://github.com/openvinotoolkit/openvino/pull/4932
+    torch_softplus = F.softplus
+    F.softplus = softplus
 
     buf = io.BytesIO()
-    inp = torch.randn(inp_shape)
     torch.onnx.export(self._torch_model.model, inp, buf, opset_version=11)
```

Comment:
We need to control input names by ourselves to prevent inconsistency

Reply:
What do you mean under control?